ConfigCreator/ConfigCreator/ConfigCreator.py

```diff
@@ -0,0 +1,544 @@
+                            hash=asset_list[keys]['hash']
+                            hash_list.append(hash)
+
+    for i in hash_list:
+        if os.path.isfile(Dot_Minecraft_Path+'/assets/objects/'+i[0:2]+'/'+i):
+            assets_list.append(Dot_Minecraft_Path+'/assets/objects/'+i[0:2]+'/'+i)
+
+    return assets_list
+
+def calculate_file_to_zip(Dot_Minecraft_Path,Full_version_dict,selected_version):
+    version_dict=Full_version_dict[selected_version]
+
+    file_list={}
+    folder_list={}
+    # Jar file and native folder
+    file_list['others']=[version_dict['Jar_file']]
+    if version_dict['Native_files_folder']:
+        folder_list['natives']=version_dict['Native_files_folder']
+
+    # Mod file
+    folder_list['mods']=Dot_Minecraft_Path+'/mods'
+
+    # Libraries file
+    file_list['libraries']=[]
+    for i in version_dict['libraries_path']:
+        file_list['libraries'].append(Dot_Minecraft_Path+'/'+i)
+
+    # Assets file
+    file_list['assets']=assets_list_analyzer(Dot_Minecraft_Path,Full_version_dict,selected_version)
+
+    # Scripts file
+    folder_list['scripts']=Dot_Minecraft_Path+'/scripts'
+
+    # Root path file
+    for i in os.listdir(Dot_Minecraft_Path):
+        if os.path.isfile(Dot_Minecraft_Path+'/'+i):
+            file_list['others'].append(Dot_Minecraft_Path+'/'+i)
+
+    # Config file
+    folder_list['config']=Dot_Minecraft_Path+'/config'
+
+    return (file_list,folder_list)
+
+def json_dump(upload_path,Dot_Minecraft_Path,Full_version_dict,selected_version,entirePackageFiles):
+    json_structure=Templete
+    startupArguments=json_structure['startupArguments']
+    startupArguments['libraryPaths']=Full_version_dict[selected_version]['libraries_path']
+    startupArguments['nativespath']='natives'
+    startupArguments['mainJarPath']=Full_version_dict[selected_version]['Jar_file'][len(Dot_Minecraft_Path)+1:]
+    startupArguments['mainClass']=Full_version_dict[selected_version]['mainClass']
+    startupArguments['version']=Full_version_dict[selected_version]['version']
+    startupArguments['assetsDir']='assets'
+    startupArguments['assetIndex']=Full_version_dict[selected_version]['assetIndex']
+    startupArguments['tweakClasses']=Full_version_dict[selected_version]['tweakClasses']
+
+    json_structure['fileSystem']['entirePackageFiles']=entirePackageFiles
+
+    json_structure['startupArguments']=startupArguments
+
+    #os.mknod(upload_path+json_structure['instanceName']+'.json')
+    json_file=open(upload_path+'/'+json_structure['instanceName']+'.json','w+')
+    json_file.write(json.dumps(json_structure, indent=4))
+    json_file.close()
+    return json_structure
+
+def zipping(Dot_Minecraft_Path,destination,file_and_folder_tuple):
+    file_list,folder_list=file_and_folder_tuple
+    entirePackageFiles=[]
+    section_sample={'downloadLink':'','md5':'','name':'','localPath':''}
+    threads=[]
+    threadLock=threading.Lock()
+
+    '''
+    class zip_file_thread (threading.Thread):
```

Comment:
It is just the problem code. I will use it after i solve the problem passing arguments to run()